tests/test_core.py

```diff
@@ -1118,6 +1120,67 @@ class MyClass(NSObject):
             class MyClass(NSObject):  # noqa: F811
                 pass
 
+    def test_class_auto_rename(self):
+        """Test automatic renaming system of ObjCClass."""
+
+        # Check that the global option works.
+
+        ObjCClass.auto_rename = True
```

Comment:
As noted above - if this is a persistent effect, then the entire test should be in a try/finally block:

```
try:
    ObjCClass.auto_rename = True

   ... do test here ...
finally:
    ObjCClass.auto_rename = None
```